src/eei.cpp

```diff
@@ -679,33 +689,28 @@ inline int64_t maxCallGas(int64_t gas) {
       call_message.code_hash = {};
       call_message.depth = msg.depth + 1;
 
-      if (kind == EEICallKind::Call || kind == EEICallKind::CallCode) {
-        valueOffset = arguments[2].geti32();
-        dataOffset = arguments[3].geti32();
-        dataLength = arguments[4].geti32();
-
+      switch (kind) {
+      case EEICallKind::Call:
+      case EEICallKind::CallCode:
         call_message.sender = msg.destination;
         call_message.value = loadUint128(valueOffset);
         call_message.kind = (kind == EEICallKind::CallCode) ? EVMC_CALLCODE : EVMC_CALL;
 
         if ((kind == EEICallKind::Call) && !isZeroUint256(call_message.value)) {
           ensureCondition(!(msg.flags & EVMC_STATIC), StaticModeViolation, "call");
         }
-      } else {
-        valueOffset = 0;
-        dataOffset = arguments[2].geti32();
-        dataLength = arguments[3].geti32();
-
-        if (kind == EEICallKind::CallDelegate) {
-          call_message.sender = msg.sender;
-          call_message.value = msg.value;
-          call_message.kind = EVMC_DELEGATECALL;
-        } else if (kind == EEICallKind::CallStatic) {
-          call_message.sender = msg.destination;
-          call_message.value = {};
-          call_message.kind = EVMC_CALL;
-          call_message.flags |= EVMC_STATIC;
-        }
+        break;
+      case EEICallKind::CallDelegate:
+        call_message.sender = msg.sender;
+        call_message.value = msg.value;
+        call_message.kind = EVMC_DELEGATECALL;
+        break;
+      case EEICallKind::CallStatic:
+        call_message.sender = msg.destination;
+        call_message.value = {};
```

Comment:
This is not needed. The `.value` is already zerod.

Reply:
I think these are done allover the code because at some point you wanted them to be explicitly initialised?

Happy to do a cleanup, but would do it in a separate PR, since there are about 20 lines affected.